### ISSUE TITLE
Argument of type 'Element' is not assignable to parameter of type 'HTMLElement'.

### ISSUE THREAD

**Issue by Semigradsky** [testing-library/jest-dom#312]
- `@testing-library/react` version: 11.2.2

### Relevant code or config:

```tsx
	const { container } = render( // container is `Element` here
		<MyComponent/>,
	)

	expect(getByText(container, someText)).toBeInTheDocument() // Error: Argument of type 'Element' is not assignable to parameter of type 'HTMLElement'.
	expect(getByTestId(container, 'test-id')).toBeInTheDocument() // Error: Argument of type 'Element' is not assignable to parameter of type 'HTMLElement'.
```

### Problem description:

After testing-library/react-testing-library#833 I get type errors because `container` no more `HTMLElement`.

`container as HTMLElement` doesn't seem like a good solution.

**Comment by gnapse** [comment]
I took the liberty to transfer this issue to jest-dom where it belongs. Seems that this needs a fix similar to that in #307.

**Comment by eps1lon** [comment]
I read the issue description again and don't understand how this would mean a runtime error. The wording doesn't match the runtime errors, sounds like a type error and matches exactly what TypeScript's type-checker says right now.

**Comment by MatanBobi** [comment]
> I read the issue description again and don't understand how this would mean a runtime error. The wording doesn't match the runtime errors, sounds like a type error and matches exactly what TypeScript's type-checker says right now.

I agree on this one, but `jest-dom` doesn't have any type definitions as far as I can see so I thought it's related to the types defined in `testing-library-dom` here:
https://github.com/testing-library/dom-testing-library/blob/c6e7a83948ef0d1822c5df1980b133ceb7e70dca/types/queries.d.ts#L57

**Comment by eps1lon** [comment]
> but jest-dom doesn't have any type definitions as far as I can see

Argh, I pasted the wrong link [previously](https://github.com/testing-library/jest-dom/issues/312#issuecomment-733764575). They do have definitions: https://www.npmjs.com/package/@types/testing-library__jest-dom

**Comment by Semigradsky** [comment]
> They do have definitions: [npmjs.com/package/@types/testing-library__jest-dom](https://www.npmjs.com/package/@types/testing-library__jest-dom)

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/testing-library__jest-dom/index.d.ts
